search.js

```diff
@@ -0,0 +1,68 @@
+"use strict";
+
+//display the list results
+var appendArticleAttributes = function(article) {
+
+  var title = document.createElement('div');
+  var section = document.createElement('div');
+  var byline = document.createElement('div');
+  var link = document.createElement('a');
+  var li = document.createElement('li');
+
+	title.textContent = "Title: " + article.title;
+  section.textContent = "Section: " + article.section;
+  byline.textContent = "Byline: " + article.byline;
+  link.textContent = "Link";
+  link.href = article.url;
+
+  li.appendChild(title);
+  li.appendChild(section);
+  li.appendChild(byline);
+  li.appendChild(link);
+
+  list.appendChild(li);
+}
+
+//make request and map over results
+var makeRequest = function() {
+
+  var searchTerm = document.getElementById("searchBox").value;
+  var request = new XMLHttpRequest();
+
+  request.open('GET', 'https://api.nytimes.com/svc/topstories/v2/science.json?api-key=Gwxln5M3geWlhR6UE0TY1FUWKSG3wCil');
+
+  request.onload = function() {
```

Comment:
How often do we think the results will be updated server side? If we don't care about having down-to-the-second realtime results we could consider retrieving all articles on page load, then manipulating the returned data upon form submission. That should increase the perception of responsiveness for the user. We could also refresh the data from the server on an interval.

Reply:
Great suggestion